src/main/java/org/prebid/server/bidder/connectad/ConnectadBidder.java

```diff
@@ -0,0 +1,199 @@
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.connectad.ExtImpConnectAd;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * ConnectAd {@link Bidder} implementation.
+ */
+public class ConnectadBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpConnectAd>> CONNECTAD_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpConnectAd>>() {
+            };
+    private static final String HTTPS_PREFIX = "https";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public ConnectadBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final int secure = secureFrom(request.getSite());
+
+        final List<BidderError> errors = new ArrayList<>();
+        final List<Imp> processedImps = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpConnectAd impExt = parseImpExt(imp);
+                final Imp updatedImp = updateImp(imp, secure, impExt.getSiteId(), impExt.getBidfloor());
+                processedImps.add(updatedImp);
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+        if (CollectionUtils.isNotEmpty(errors)) {
+            errors.add(BidderError.badInput("Error in preprocess of Imp"));
+            return Result.of(Collections.emptyList(), errors);
+        }
+        final BidRequest outgoingRequest = request.toBuilder().imp(processedImps).build();
+
+        final String body = mapper.encode(outgoingRequest);
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<BidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(endpointUrl)
+                        .headers(resolveHeaders(request.getDevice()))
+                        .payload(outgoingRequest)
+                        .body(body)
+                        .build()),
+                errors);
+    }
+
+    private int secureFrom(Site site) {
+        final String page = site != null ? site.getPage() : null;
+        return StringUtils.isNotBlank(page) && page.startsWith(HTTPS_PREFIX) ? 1 : 0;
```

Comment:
i see the code is absolutely right but [LGTM](https://lgtm.com/projects/g/rubicon-project/prebid-server-java/rev/pr-0600e194ed59050f73d2264ddfacf2f9e599840b) thinks different.
May you pls change `StringUtils.isNotBlank(page)` to plain `page != null`?